### PR TITLE
create a table of contents

### PR DESCRIPTION
**Pull request by nikomatsakis**
cc @detrumi @jackh726

### CODE REVIEW

**Comment by nikomatsakis** [comment]
I want to put a bit more notes into the files to elaborate more what I think they would say

File: book/src/SUMMARY.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
 # Summary
```
Comment (jackh726) [inline book/src/SUMMARY.md]:
```diff
- # Summary
+ # Table of Contents
```

File: book/src/crates.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Create breakdown
```
Comment (lqd) [inline book/src/crates.md]:
```suggestion
# Crate breakdown
```

Comment (nikomatsakis) [reply book/src/crates.md]: whoops, yeah, I shouldn't have added this

File: book/src/what_is_chalk/crates.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+Chalk is broken up into a number of crates. This chapter explains the
+role of each crate. This crate structure helps to serve Chalk's two goals:
+
+* To serve as the trait engine for compilers and tools like rustc and rust-analyzer
+* To be usable as a standalone REPL and testing harness
+
+## Crates for embedding chalk into other programs
+
+## Crates for
```
Comment (jackh726) [inline book/src/what_is_chalk/crates.md]: Duplicate line

File: book/src/what_is_chalk/crates.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+## Crates for 
+
+Although there are many crates, there are two main "entry-point"
+crates that define the different "levels" of chalk:
+
+* the main `chalk` crate, along with the testing crate in the `tests` directory
+    * These define the REPL and 
+* the `chalk-solve` crate
+* `chalk-integration`: The highest level crate. This is the "testing
```
Comment (jackh726) [inline book/src/what_is_chalk/crates.md]: Formatting? Ident?

File: book/src/what_is_chalk/crates.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+## Crates for embedding chalk into other programs
+
+## Crates for 
+
+Although there are many crates, there are two main "entry-point"
+crates that define the different "levels" of chalk:
+
+* the main `chalk` crate, along with the testing crate in the `tests` directory
+    * These define the REPL and
```
Comment (jackh726) [inline book/src/what_is_chalk/crates.md]: and ...

File: book/src/what_is_chalk/crates.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+* the main `chalk` crate, along with the testing crate in the `tests` directory
+    * These define the REPL and 
+* the `chalk-solve` crate
+* `chalk-integration`: The highest level crate. This is the "testing
+  harness" and is also used by the REPL.
+* `chalk-solve`: The mid-level crate where 
+* `chalk-engine`: This is the lowest-level crate.
+
+being both embeddable into compilers like rustc and rust-analyzer,
```
Comment (jackh726) [inline book/src/what_is_chalk/crates.md]: Typo?

**Comment by nikomatsakis** [comment]
cleaned up the "crates" PR

File: book/src/what_is_chalk/crates.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+other programs:
+
+* The `chalk-solve` crate, which defines the rules that translate Rust IR into logical predicates.
+* The `chalk-ir` crate, which defines the IR representing types and logical predicates.
+* The `chalk-rust-ir` crate, which defines the IR representing Rust concepts like traits and impls.
+
+The following crate is an implementation detail, used internally by `chalk-solve`:
+
+* The `chalk-engine` crate, which defines the actual engine that solves logical predicate. This 
+  engine is quite general and not really specific to
```
Comment (jackh726) [inline book/src/what_is_chalk/crates.md]: Specific to Rust?

**Comment by jackh726** [comment]
Just one minor typo, but I think overall this is a good start!

**Pull request merged**